pennylane/_device.py

```diff
@@ -273,10 +273,11 @@ def __exit__(self, type, value, traceback):
         return MockContext()
 
     def supports_operation(self, operation):
-        """Checks if an operation is supported by this device.
+        """Checks if an operation is supported by this device. Raises a ValueError,
+         if not a subclass or string of an Operation was passed.
```

Comment:
Instead of including the second line in the docstring here, better to add a `Raises:` section below the `Returns:` section, like so:
```rest
    Raises:
        ValueError: if `operation` is not a :class:`~.Operation` class or string
```

This will be picked up by Sphinx in the documentation. In general, we should do a better job of documenting exceptions, we very rarely end up using the `Raises:` section in PennyLane!

Reply:
That's a really good idea, thanks for that!

Reply:
Added the quick PR #400 for that